libraries/AP_Camera/examples/tracking.py

```diff
@@ -0,0 +1,183 @@
+import cv2
+import socket
+import struct
+import threading
+import math
+from pymavlink import mavutil
+import time
+
+class GimbalControl:
+    def __init__(self, connection_str):
+        self.master = mavutil.mavlink_connection(connection_str)
+        self.master.wait_heartbeat()
+        print("Heartbeat from the system (system %u component %u)" % 
+              (self.master.target_system, self.master.target_component))
+        self.center_x = 0
+        self.center_y = 0
+        self.lock = threading.Lock()  # Initialize a lock for shared variables
+        self.control_thread = threading.Thread(target=self.send_command)
+        self.control_thread.start()
+
+    def send_gimbal_manager_pitch_yaw_angles(self, pitch, yaw, pitch_rate, yaw_rate):
+        msg = self.master.mav.gimbal_manager_set_pitchyaw_encode(
+            self.master.target_system,
+            self.master.target_component,
+            0,
+            0,
+            pitch,
+            yaw,
+            pitch_rate,
+            yaw_rate
+        )
+        self.master.mav.send(msg)
+
+    def send_command(self):
+        while True:
+            start_time = time.time()  # Record the start time of the loop
+            
+            with self.lock:  # Lock when accessing shared variables
+                centre_x_copy = int(self.center_x)
+                centre_y_copy = int(self.center_y)
+
+            if (centre_x_copy == 0 and centre_y_copy == 0):
+                diff_x = 0
+                diff_y = 0
+            else:
+                diff_x = (centre_x_copy - (640 / 2)) / 2
+                diff_y = -(centre_y_copy - (480 / 2)) / 2
+
+            self.send_gimbal_manager_pitch_yaw_angles(float("NaN"), float("NaN"), math.radians(diff_y), math.radians(diff_x))
```

Comment:
I think we're going to need PI(D) controllers to set gains on the conversion from `(diff_x, diff_y)` to `(pitch_rate_rads, yaw_rate_rads)`. From initial testing with a SIYI A8, P gains of around 0.1 are a reasonable start. Unadjusted, the gimbal overshoots and tracking is lost immediately.